File: spk/Client/src/models/userSettings.ts
export enum UserThemePreference {
  Light = 1,
  Dark = 2,
  System = 3,
}

export enum UserLanguagePreference {
  Turkish = 1,
  English = 2,
}

export enum UserDateFormatPreference {
  DayMonthYear = 1,
  YearMonthDay = 2,
  MonthDayYear = 3,
}

export enum UserTimeFormatPreference {
  TwentyFourHour = 1,
  TwelveHour = 2,
}

export enum UserDefaultQuizMode {
  Mixed = 1,
  Topic = 2,
  Course = 3,
  WrongAnswers = 4,
  TrialExam = 5,
}

export enum UserQuestionTransitionPreference {
  Manual = 1,
  AfterAnswer = 2,
  AfterCorrectAnswer = 3,
}

export enum UserPdfViewPreference {
  Pdf = 1,
  Text = 2,
  Split = 3,
}

export interface UserSettings {
  userId: string
  theme: UserThemePreference
  language: UserLanguagePreference
  dateFormat: UserDateFormatPreference
  timeFormat: UserTimeFormatPreference
  compactView: boolean
  emailNotifications: boolean
  pushNotifications: boolean
  newContentNotifications: boolean
  trialReminders: boolean
  reviewReminder: boolean
  dailyGoalReminder: boolean
  weeklyGoalReminder: boolean
  studyReminders: boolean
  supportTicketUpdates: boolean
  dailyGoalQuestionCount: number
  dailyStudyMinutes: number
  examDate?: string | null
  weeklyStudyMinutes: number
  preferredStudyDays: string
  defaultQuizMode: UserDefaultQuizMode
  defaultQuestionCount: number
  autoReviewSuggestions: boolean
  timedQuizDefault: boolean
  defaultQuizDurationMinutes: number
  autoOpenExplanations: boolean
  questionTransition: UserQuestionTransitionPreference
  autoAddWrongAnswersToReview: boolean
  preferredPdfView: UserPdfViewPreference
  rememberLastPdfPage: boolean
  autoSaveNotes: boolean
  showHighlights: boolean
  securityNotifications: boolean
  createdAt: string
  updatedAt?: string | null
}

export type UpdateUserSettings = Omit<UserSettings, 'userId' | 'createdAt' | 'updatedAt'>

export interface UserSecuritySettings {
  email: string
  displayName: string
  emailConfirmed: boolean
  twoFactorReady: boolean
  activeSessionCount: number
  refreshTokenExpiresAt?: string | null
  createdAt: string
  securityNotifications: boolean
}

export const themeOptions = [
  { value: UserThemePreference.Light, label: 'Açık Tema' },
  { value: UserThemePreference.Dark, label: 'Koyu Tema' },
  { value: UserThemePreference.System, label: 'Sistem Teması' },
] as const

export const languageOptions = [
  { value: UserLanguagePreference.Turkish, label: 'Türkçe' },
  { value: UserLanguagePreference.English, label: 'English' },
] as const

export const dateFormatOptions = [
  { value: UserDateFormatPreference.DayMonthYear, label: 'GG.AA.YYYY' },
  { value: UserDateFormatPreference.YearMonthDay, label: 'YYYY-AA-GG' },
  { value: UserDateFormatPreference.MonthDayYear, label: 'AA/GG/YYYY' },
] as const

export const timeFormatOptions = [
  { value: UserTimeFormatPreference.TwentyFourHour, label: '24 saat' },
  { value: UserTimeFormatPreference.TwelveHour, label: '12 saat' },
] as const

export const quizModeOptions = [
  { value: UserDefaultQuizMode.Mixed, label: 'Karışık Test' },
  { value: UserDefaultQuizMode.Topic, label: 'Alt Konu Testi' },
  { value: UserDefaultQuizMode.Course, label: 'Ders Bazlı Test' },
  { value: UserDefaultQuizMode.WrongAnswers, label: 'Yanlışlarım' },
  { value: UserDefaultQuizMode.TrialExam, label: 'Deneme' },
] as const

export const questionTransitionOptions = [
  { value: UserQuestionTransitionPreference.Manual, label: 'Manuel' },
  { value: UserQuestionTransitionPreference.AfterAnswer, label: 'Cevaptan sonra otomatik' },
] as const

export const pdfViewOptions = [
  { value: UserPdfViewPreference.Pdf, label: 'PDF' },
  { value: UserPdfViewPreference.Text, label: 'Metin' },
  { value: UserPdfViewPreference.Split, label: 'Bölünmüş görünüm' },
] as const

File: spk/API/Entities/UserSettings.cs
namespace API.Entities;

public class UserSettings : BaseEntity
{
    public string UserId { get; set; } = string.Empty;

    public AppUser? User { get; set; }

    public UserThemePreference Theme { get; set; } = UserThemePreference.System;

    public UserLanguagePreference Language { get; set; } = UserLanguagePreference.Turkish;

    public UserDateFormatPreference DateFormat { get; set; } = UserDateFormatPreference.DayMonthYear;

    public UserTimeFormatPreference TimeFormat { get; set; } = UserTimeFormatPreference.TwentyFourHour;

    public bool CompactView { get; set; }

    public bool EmailNotifications { get; set; } = true;

    public bool PushNotifications { get; set; } = true;

    public bool NewContentNotifications { get; set; } = true;

    public bool TrialReminders { get; set; } = true;

    public bool ReviewReminder { get; set; } = true;

    public bool DailyGoalReminder { get; set; } = true;

    public bool WeeklyGoalReminder { get; set; } = true;

    public bool StudyReminders { get; set; } = true;

    public bool SupportTicketUpdates { get; set; } = true;

    public int DailyGoalQuestionCount { get; set; } = 25;

    public int DailyStudyMinutes { get; set; } = 45;

    public DateTime? ExamDate { get; set; }

    public int WeeklyStudyMinutes { get; set; } = 315;

    public string PreferredStudyDays { get; set; } = "1,2,3,4,5,6,0";

    public UserDefaultQuizMode DefaultQuizMode { get; set; } = UserDefaultQuizMode.Mixed;

    public int DefaultQuestionCount { get; set; } = 20;

    public bool AutoReviewSuggestions { get; set; } = true;

    public bool TimedQuizDefault { get; set; }

    public int DefaultQuizDurationMinutes { get; set; } = 30;

    public bool AutoOpenExplanations { get; set; } = true;

    public UserQuestionTransitionPreference QuestionTransition { get; set; } = UserQuestionTransitionPreference.Manual;

    public bool AutoAddWrongAnswersToReview { get; set; } = true;

    public UserPdfViewPreference PreferredPdfView { get; set; } = UserPdfViewPreference.Pdf;

    public bool RememberLastPdfPage { get; set; } = true;

    public bool AutoSaveNotes { get; set; } = true;

    public bool ShowHighlights { get; set; } = true;

    public bool SecurityNotifications { get; set; } = true;
}
